includes/wikia/api/DesignSystemApiController.class.php

```diff
@@ -195,4 +203,57 @@ private function isWikiaOrgCommunity() {
 
 		return wfGetBaseDomainForHost( parse_url( $wgServer, PHP_URL_HOST ) ) === $wgWikiaOrgBaseDomain;
 	}
+
+
+	/**
+	 * External API request to IntentX
+	 */
+	public function getFandomShopDataFromIntentX() {
+		global $wgCityId, $wgMemc, $wgFandomShopMap, $wgFandomShopUrl;
+
+		$key = self::MEMC_PREFIX_FANDOM_STORE;
+		// get id from parameter or default to city id
+		$id = intval( $this->getVal( static::PARAM_ID, $wgCityId) );
+		// not using wfMemcKey here to avoid prefix conflicts
+		$memcKey = "intentX:$key:$id";
+
+		// don't make api call if not fandom store community
+		if ( !array_key_exists( $id, $wgFandomShopMap ) ) {
+			return null;
+		}
+
+		// do api request
+		$client = new Client( [
+			'base_uri' => $wgFandomShopUrl,
+			'timeout' => 30.0
+		] );
+		$params = [
+			'clientId' => 'fandom',
+			'relevanceKey' => $wgFandomShopMap[ $id ],
+		];
+
+		try {
+			$response = $client->get( '', [
+				'query' => build_query( $params, PHP_QUERY_RFC1738 ),
+			] );
+			$body = $response->getBody();
+			$data = json_decode( $body );
+			// store in cache indefinitely
+			$wgMemc->set( $memcKey, $data, self::TTL_INFINITE );
```

Comment:
I'm a little confused here. So this controller method always makes the request and never reads the cached result? Is that intended?